### DISCUSSION TITLE
Cannot read property 'zone' of undefined

### DISCUSSION THREAD

**Discussion by TheArKaID** [adonisjs/core#2719] [Help] [answered]
I got this error when creating a Middleware (called Front), add it to the Kernel, and then in the Middleware I try to access Date:
```js
public async handle ({ session }: HttpContextContract, next: () => Promise<void>) {
    let visitDate = session.get('visit-date', false)
    let now = new DateTime()
    console.log(now)

    // code for middleware goes here. ABOVE THE NEXT CALL
    await next()
  }
```

And get this error " Cannot read property 'zone' of undefined ". In the error stack, there's Luxon package that throw the error,
```js
constructor(config) {
    const zone = config.zone || Settings.defaultZone;

    let invalid =
      config.invalid ||
      (Number.isNaN(config.ts) ? new Invalid("invalid input") : null) ||
      (!zone.isValid ? unsupportedZone(zone) : null);
}
```

I thought that I missed one of the configuration in the `config` directory, but I don't know which one is that.

**Comment by thetutlage** [comment, marked as answer]
The usage of luxon DateTime object is incorrect. It must be like this https://moment.github.io/luxon/docs/class/src/datetime.js~DateTime.html#static-method-local

**Reply by TheArKaID** [reply]
```js
import { DateTime } from 'luxon'

let now = DateTime.local().toISODate() // 2021-06-02
```

Thanks
